Dockerfile

```diff
@@ -0,0 +1,28 @@
+# Build Hapistrano
+FROM fpco/stack-build as build-env
+
+MAINTAINER Javier Casas <jcasas@stackbuilders.com>
+
+WORKDIR /hapistrano
+COPY stack.yaml .
+COPY hapistrano.cabal .
+RUN stack build --only-dependencies
```

Comment:
@javcasas in order to download the dependencies you only need `hapistrano.cabal` what about moving `COPY stack.yaml .` below `RUN stack build --only-dependencies` in order to preserve the cache.